src/lib/common/sol-missing.h

```diff
@@ -70,6 +70,39 @@ memmem(const void *haystack, size_t haystacklen, const void *needle, size_t need
 }
 #endif
 
+#if !defined(HAVE_GET_CURRENT_DIR_NAME) && defined(HAVE_UNIX)
```

Comment:
Note this depends on HAVE_UNIX being defined, which is in another patch not yet ready for submission. But this won't break the build.